packages/next/src/client/use-intersection.tsx

```diff
@@ -127,7 +127,7 @@ export function useIntersection<T extends Element>({
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isDisabled, rootMargin, rootRef, visible, elementRef.current])
+  }, [isDisabled, rootMargin, rootRef, visible])
```

Comment:
While the ref is stable, current is not.
```suggestion
  }, [isDisabled, rootMargin, rootRef, visible, elementRef.current])
```